### DISCUSSION TITLE
How to implement payment management in Smart contract

### DISCUSSION THREAD

**Discussion by ariesgun** [aptos-labs/aptos-developer-discussions#415] [Questions]
### Describe your question in detail.

Is there any good example on how to manage payment in Smart Contract?
I wonder if I should use `coin` or `FA`.

Questions:
- How to find the address of APT token object or coin?
- How to find the address of USDC token object FA?

Use cases:
- Users send APT tokens and in return, the SC sends an NFT.
- Users send USDC tokens and the SC sends an NFT.

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by kshitijc1506** [comment]
1. APT FA metadata object address is 0xA
2. I believe zUSDC / whUSDC is Coin standard -- here's an explorer [link](https://explorer.aptoslabs.com/account/0xf22bede237a07e121b56d91a491eb7bcdfd1f5907926a9e58338f964a01b17fa/resources?network=mainnet) with relevant info 

Ideally, since we're in the middle of migration, you'd support both Coin and FA. Find more details about the [migration](https://aptos.dev/en/build/smart-contracts/fungible-asset#migration-from-coin-to-the-fungible-asset-standard) in the docs. 

Although, you want to verify if there's sufficient liquidity or CEX support before converting coins to FA on mainnet.
